content/blog/excel-to-r.md

```diff
@@ -0,0 +1,72 @@
+---
+title: "Experiments in transforming Excel into R"
+date: 2019-11-18
+---
+
+Last week was the [Health Economics in R Hackathon](https://n8thangreen.wixsite.com/hermes-hack-london), bringing together health economists, epidemiologists and software engineers.
+
+We learned that complex spreadsheet models are common in health economics, and that the users of these models feel constrained by what they can achieve with them - having constructed a [decision tree model](https://en.wikipedia.org/wiki/Decision_tree) in Excel, the analyst might want to carry out a [sensitivity analysis](https://en.wikipedia.org/wiki/Sensitivity_analysis) vary the models' parameters seeing how the outcomes vary.  Doing this requires knowledge of [Visual Basic](https://en.wikipedia.org/wiki/Visual_Basic_for_Applications) and reportedly is quite slow to run.  We wanted to create a proof-of-concept package that would take an excel model like this and convert it into an R function, where it could be more easily manipulable.
+
+This does not present a complete solution by any means (it _was_ a hackathon)!  But it's a proof-of-concept for an idea that might be worth pursuing.
+
+First, it's worth thinking about what it *means* to "compute a spreadsheet".  In our brainstorming session we tried to home in on the problem - what is the analyst actually trying to do, and how would they do it?  The basic idea is that they would modify some cells that represent parameters and then the spreadsheet recomputes ([spreadsheets fundamentally representing a form of reactive programming](https://stenci.la/blog/introducing-sheets/)) to change other cells that represent inputs.  So we have a mapping of inputs to outputs, which seems an awful lot like a function.
+
+While clever approaches could be developed to automatically determine inputs and outputs, we allowed specifying of inputs and outputs as a set of Excel cell references (using [`cellranger`](https://github.com/rsheets/cellranger)) on a particular sheet, with a hint as to where to find a label as a row/column offset:
+
+```
+inputs <- xlerate::xlerate_ref(
```

Comment:
Could make *much* more of the xlerate pun than just letting it introduce itself in some code! :)

Reply:
Is it pronounced "accelerate", or "exhilerate" - both are very worthy descriptions.

Reply:
I've done what I can but finding a lot of punsperation today!